metabase/metabase
# remove static viz import from pie chart option

Thread https://metaboat.slack.com/archives/C505ZNNH4/p1723203837352799?thread_ts=1723201440.582679&cid=C505ZNNH4

### Description

Importing `truncateText` from the static viz directory caused our bundle size to increase significantly. In this PR we remove that import, and replace it with an import for a similar function in the main app directory. We had to create a similar `truncateText` function from scratch since one didn't already exist.

### How to verify

Create a pie chart where dimension values have long names, or with a long prefix/suffix for the metric in column settings, add to a dashboard on a small dashcard, hover a slice or legend item and confirm truncation still works.

### Demo

<img width="417" alt="Screenshot 2024-08-22 at 6 50 33 PM" src="https://github.com/user-attachments/assets/9bc7b39b-3aa7-4378-841d-f722e3ae72af">

### Checklist


Covered by existing storybook test for truncation.
